h2o-docs/src/product/data-munging/target-encoding.rst

```diff
@@ -1,461 +1,160 @@
-
-        auc_comparison
-                               Model       AUC
-        1         No Target Encoding 0.7070187
-        2              No addr_state 0.7076197
-        3 addr_state Target Encoding 0.7072750
+    library(h2o)
+    h2o.init()
+
+    #Import the titanic dataset
+    f <- "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+    titanic <- h2o.importFile(f)
+     
+    # Set response column as a factor
+    response <- "survived"
+    titanic[response] <- as.factor(titanic[response])
+     
+    # Split the dataset into train and test
+    splits <- h2o.splitFrame(data = titanic, ratios = .8, seed = 1234)
+    train <- splits[[1]]
+    test <- splits[[2]]
+     
+    # Choose which columns to encode
+    encoded_columns <- c("home.dest", "cabin", "embarked")
+     
+    # Train a TE model
+    te_model <- h2o.targetencoder(x = encoded_columns,
+                                  y = response, 
+                                  training_frame = train,
+                                  fold_column = "pclass", 
+                                  data_leakage_handling = "KFold")
+
+    # New target encoded train and test sets
+    train_te <- h2o.transform(te_model, train)
+    test_te <- h2o.transform(te_model, test)
 
    .. code-tab:: python
 
-        # Compare AUC values:
+    library(h2o)
+    h2o.init()
+    from h2o.estimators import H2OTargetEncoderEstimator
 
-        valid_auc = gbm_baseline.auc(valid=True)
-        valid_auc
-        0.707018686126265
-
-        auc_nostate = gbm_no_state.auc(valid=True)
-        auc_nostate
-        0.7076197256885596
-
-        auc_state_te = gbm_state_te.auc(valid=True)
-        auc_state_te
-        0.7072749724799465
+    #Import the titanic dataset
+    titanic = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv")
+    
+    # Set response column as a factor
+    titanic['survived'] = titanic['survived'].asfactor()
+    response='survived'
 
-Now the ``addr_state_te`` has much smaller variable importance. It is no longer the second most important feature but the 10th.
+    # Split the dataset into train and test
+    train, test = titanic.split_frame(ratios = [.8], seed = 1234)
 
-.. tabs::
-   .. code-tab:: r R
+    # Choose which columns to encode
+    encoded_columns = ["home.dest", "cabin", "embarked"]
 
-        # Variable Importance
-        h2o.varimp_plot(gbm_state_te)
+    # Set target encoding parameters
+    fold_column = "pclass"
+    data_leakage_handling = "k_fold"
 
-   .. code-tab:: python
+    # Train a TE model
+    titanic_te = H2OTargetEncoderEstimator(fold_column=fold_column,
+                                           data_leakage_handling=data_leakage_handling)
+    te_model = titanic_te.train(x=encoded_columns, 
+                                y=response,
+                                training_frame=titanic)
```

Comment:
@angela0xdata @ledell  we should not use whole `titanic` dataset for training. I'm working right now on the extended demo ( with comparison) . Not sure if I will make it in time, but I will try.

Reply:
@angela0xdata @ledell found optimal values, prepared tests ( for both an old api and a new api as I needed to compute aucs with the old api due to a bug in the new one). Transferring code to .rst file.